fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AnalyzerUtils.java

```diff
@@ -1412,11 +1419,26 @@ public static AddPartitionClause getAddPartitionClauseFromPartitionValues(OlapTa
                     partitionName = partitionNamePrefix + PARTITION_NAME_PREFIX_SPLIT + partitionName;
                 }
                 if (!partitionColNames.contains(partitionName)) {
+                    List<List<String>> partitionItems = Collections.singletonList(partitionValue);
+                    PListCell cell = new PListCell(partitionItems);
+                    // If partition name already exists and their partition values are not the same, change partition name.
+                    if (tablePartitions.containsKey(partitionName) && !tablePartitions.get(partitionName).equals(cell)) {
+                        // change partition name, how to generate a unique partition name
+                        int diff = calculateStringDiff(partitionName, partitionName.toLowerCase(Locale.ROOT));
+                        partitionName = partitionName + "_" + Integer.toHexString(diff);
```

Comment:
what's the purpose to add the diff into partition name ? 
is there any compatibility issue with it ?

Reply:
partiiton names in FE  are stored as case-insensitive, we should ensure its name should be different in  case-insensitive.

You can see current bug in : https://github.com/StarRocks/starrocks/issues/54202
```
    @SerializedName(value = "idToPartition")
    protected Map<Long, Partition> idToPartition = new HashMap<>();
    protected Map<String, Partition> nameToPartition = Maps.newTreeMap(String.CASE_INSENSITIVE_ORDER)
```